includes/class-wc-payments-account.php

```diff
@@ -900,23 +960,138 @@ public function maybe_handle_onboarding() {
 			return;
 		}
 
+		/**
+		 * ==================
+		 * Handle connect links that trigger either the onboarding flow start or its continuation.
+		 *
+		 * "Onboarding flow" = the Jetpack connection + onboarding wizard + Stripe KYC.
+		 *
+		 * IMPORTANT: Connect links are NOT THE SAME as Connect page links:
+		 *    - Connect links are used to start/re-start/continue the onboarding flow.
+		 *      Connect links are WP admin links with the `wcpay-connect` GET parameter and a 'wcpay-connect' action nonce.
+		 *      Connect links imply actions being taken on the merchant store and account setup (hence the need to be protected by a nonce).
+		 *    - Connect page links are just the WC admin URL of the Connect page (page=wc-admin&path=/payments/connect).
+		 *
+		 * The BUSINESS LOGIC of handling connect links, in the order of priority:
+		 *
+		 * 1. Returning from the WPCOM/Jetpack connection screen.
+		 *      1.1 SUCCESSFUL connection
+		 *          1.1.1 NO Stripe account connected -> redirect to ONBOARDING WIZARD
+		 *          1.1.2 Stripe account connected -> redirect to OVERVIEW PAGE
+		 *      1.2 UNSUCCESSFUL connection -> redirect to CONNECT PAGE with ERROR message
+		 * 2. Working WPCOM/Jetpack connection and fully onboarded Stripe account -> redirect to OVERVIEW PAGE
+		 * 3. Specific `from` places -> redirect to CONNECT PAGE regardless of the account status
+		 * 4. NO [working] WPCOM/Jetpack connection:
+		 *    Initialize the WPCOM registration and:
+		 *      4.1 On SUCCESS -> redirect to WPCOM/Jetpack connection screen (Calypso).
+		 *      4.2 On ERROR -> redirect to CONNECT PAGE with ERROR message
+		 * 5. Working WPCOM/Jetpack connection:
+		 *    5.1 Handle test to live account switch -> redirect to ONBOARDING WIZARD
+		 *    5.2 Handle onboarding/account reset -> redirect to CONNECT PAGE
+		 *    5.3 If NO Stripe account connected:
+		 *         5.3.1 If we come from ONBOARDING WIZARD or use the SANDBOX MODE/BUILDER FLOW:
+		 *                Initialize the Stripe account and:
+		 *                5.3.1.1 On SUCCESS -> redirect to STRIPE KYC
+		 *                5.3.1.2 On existing account -> redirect to OVERVIEW PAGE
```

Comment:
how can this happen? We don't have Stripe account connected

Reply:
On our platform, if there is already a Stripe account associated/connected with the blog (regardless of the client state) we will simply return that back in the form of `url:false` in the response.